modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java

```diff
@@ -120,9 +120,10 @@ public static void handleErrorInboundPath(CarbonMessage cMsg, CarbonCallback cal
         String errorMsg = ErrorHandlerUtils.getErrorMessage(throwable);
         String stacktrace = ErrorHandlerUtils.getServiceStackTrace(balContext, throwable);
         String errorWithTrace = errorMsg + "\n" + stacktrace;
-        log.error(errorWithTrace);
         outStream.println(errorWithTrace);
 
+        // bre log should contain bre stack trace, not the ballerina stack trace
+        breLog.error("error : " + errorMsg + ", ballerina service stack trace : " + stacktrace, throwable);
```

Comment:
Ideally a space is not needed before the colon:
https://www.grammarly.com/handbook/punctuation/colon/